first-party/jib-quarkus-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/quarkus/JibQuarkusExtension.java

```diff
@@ -44,61 +48,65 @@
 import org.gradle.api.artifacts.ResolvedArtifact;
 import org.gradle.api.artifacts.component.ProjectComponentIdentifier;
 import org.gradle.api.plugins.JavaPlugin;
-import org.gradle.jvm.tasks.Jar;
 
-public class JibQuarkusExtension implements JibGradlePluginExtension<Void> {
+public class JibQuarkusExtension implements JibGradlePluginExtension<Configuration> {
 
   private AbsoluteUnixPath appRoot = AbsoluteUnixPath.get("/app");
   private List<String> jvmFlags = Collections.emptyList();
 
+  private JarResolver jarResolver = new LegacyJarResolver();
+  private PackageType packageType = PackageType.LEGACY;
+
   @Override
-  public Optional<Class<Void>> getExtraConfigType() {
-    return Optional.empty();
+  public Optional<Class<Configuration>> getExtraConfigType() {
+    return Optional.of(Configuration.class);
```

Comment:
Since package type is a simple string key/value, the full configuration setup is not needed. Instead, you can use the `properties` string map that's already passed in to `JibQuarkusExtension.extendContainerBuildPlan`. It would also be good to make the property name somewhat consistent with `quarkus.package.type=legacy-jar`. There is an example [here](https://github.com/GoogleContainerTools/jib-extensions/blob/5396149fbcc9c98846e798e7477cc0c8d413b76e/first-party/jib-spring-boot-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/springboot/JibSpringBootExtension.java#L87), and a longer explanation in the [readme](https://github.com/GoogleContainerTools/jib-extensions#gradle).

If you think, however, that this configuration will grow more complex over time, then it's fine to keep the `Configuration` class. In that case, could you add javadoc describing sample configuration, similar to [javadoc here](https://github.com/GoogleContainerTools/jib-extensions/blob/f267d830f859e969af0f30f4ec655cbe10f5b4fb/first-party/jib-ownership-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/ownership/Configuration.java#L27-L48)?

The configuration I've tested successfully was:
```
pluginExtensions {
        pluginExtension {
            implementation = 'com.google.cloud.tools.jib.gradle.extension.quarkus.JibQuarkusExtension'
            configuration {
                packageType {
                    name = 'FAST'
                }
            }
        }
    }
```

If you go the simple string map route, it will instead be something like:
```
pluginExtensions {
        pluginExtension {
            implementation = 'com.google.cloud.tools.jib.gradle.extension.quarkus.JibQuarkusExtension'
            properties = [packageType: 'legacy-jar']
        }
    }
```

Reply:
Great, I didn't think of that. I think that would be much better since it is simpler and I do not believe the configuration will be much more complex in the future.

Reply:
It helped with test coverage, too -- less boilerplate to cover.